Adding HIPAA retention msg for audit logs (#945)
wandb/docs: docs/guides/hosting/monitoring-usage/audit-logging.md

title: Track user activity with audit logs
---

> Use W&B audit logs to track user activity within your organization, and to conform to your enterprise governance requirements. The audit logs are JSON-formatted, and their access mechanism depends on your W&B Server deployment type:

Reviewer: ```suggestion
Use W&B audit logs to track user activity within your organization, and to conform to your enterprise governance requirements. Audit logs are formatted using JSON. How to access audit logs depends on your W&B Server deployment:
```

| W&B Server Deployment type | Audit logs access mechanism(s) |
|----------------------------|--------------------------------|
| Self-managed | Synced to instance-level bucket every 10 minutes. Also available using [the API](#fetch-audit-logs-using-api). |
| Dedicated Cloud with [secure storage connector (BYOB)](../data-security/secure-storage-connector.md) | Synced to instance-level bucket (BYOB) every 10 minutes. Also available using [the API](#fetch-audit-logs-using-api). |
| Dedicated Cloud with W&B managed storage (without BYOB) | Only available using [the API](#fetch-audit-logs-using-api). |

> **Audit logs are not available for [SaaS Cloud](../hosting-options/saas_cloud.md) yet.**

Reviewer: ```suggestion
:::info
Audit logs are not available for [SaaS Cloud](../hosting-options/saas_cloud.md).
```
Converting to an admonition, and trying to avoid future-facing language and promises. I wonder if it would be good to add something like "To express interest, contact your account team."

Once you've access to your audit logs, analyze those using your preferred tools, such as [Pandas](https://pandas.pydata.org/docs/index.html), [Amazon Redshift](https://aws.amazon.com/redshift/), [Google BigQuery](https://cloud.google.com/bigquery), [Microsoft Fabric](https://www.microsoft.com/en-us/microsoft-fabric), and more. You may need to transform the JSON-formatted audit logs into a format relevant to the tool before analysis. Information on how to transform your audit logs for specific tools is outside the scope of W&B documentation.

:::tip
**Audit Log Retention:** If a compliance, security or risk team in your organization requires audit logs to be retained for a specific period of time, W&B recommends to periodically transfer the logs from your instance-level bucket to a long-term retention storage. If you're instead using the API to access the audit logs, you can implement a simple script that runs periodically (like daily or every few days) to fetch any logs that may have been generated since the time of the last script run, and store those in a short-term storage for analysis or directly transfer to a long-term retention storage.

> If you are using a HIPAA-compliant [Dedicated Cloud](../hosting-options/dedicated_cloud.md) instance for W&B Models or Weave, you must retain the audit logs for at least 6 years. It is a regulatory requirement. You must configure guardrails for your managed storage such that an internal or external user is not able to delete the audit logs before the end of the mandatory retention period.

Reviewer: ```suggestion
HIPAA compliance requires that you retain audit logs for a minimum of 6 years. For [Dedicated Cloud](../hosting-options/dedicated_cloud.md) instances for W&B Models or Weave, you must configure guardrails for your managed storage to ensure that no internal or external user can delete audit logs before the end of the mandatory retention period.
```
